tests/simulation/test_particle_tracker.py

```diff
@@ -648,6 +653,171 @@ def test_particle_tracker_Bethe_warning(
+#     ):
+#         simulation.add_scattering("Al-27 0+", lambda: 42, "Lorem ipsum")
+
+
+PARTICLES_PER_ENERGY = 100
+
+
+@pytest.mark.parametrize(
+    (
+        "target_material",
+        "target_density",
+        "target_temperature",
+        "NIST_material_string",
+        "beam_energies",
+        "stopping_ranges",
+        "expected_lateral_scattering",
+    ),
+    [
+        (
+            "Al-27",
+            [
+                2.69890e00,
+            ]
+            * u.g
+            / u.cm**3,
+            298 * u.K,
+            "ALUMINUM",
+            [1, 5, 10] * u.MeV,
+            [16.33, 215.93, 709.23] * u.um,
+            [0.9972, 10.29, 31.70] * u.um,
+        )
+    ],
+)
+@particle_input
+@validate_quantities
+def test_particle_tracker_ranges_straggling(
+    target_material: ParticleLike,
+    target_density: u.Quantity[u.kg / u.m**3],
+    target_temperature: u.Quantity[u.K],
+    NIST_material_string: str,
+    beam_energies: u.Quantity[u.J],
+    stopping_ranges: u.Quantity[u.m],
+    expected_lateral_scattering: u.Quantity[u.m],
+):
+    """
+    Test both the stopping range and scattering distribution of the particle
+    tracker for a variety of materials and energies.
+
+    """
+    # TODO: add implementation-specific details to this docstring
+
+    beam_particle = Particle("p+")
+
+    # Let the width of the grid be dynamic between materials based on the
+    # maximum stopping range
+    width = np.max(stopping_ranges)
+    grid = CartesianGrid(
+        [-0.2, 0.0, -0.2] * u.cm, [0.2, width.to(u.cm).value, 0.2] * u.cm
+    )
+
+    # Construct the quantity arrays for the grid
+    grid_shape_identity = np.ones(grid.shape)
+    temperature = grid_shape_identity * target_temperature
+    grid.add_quantities(T_i=temperature)
+
+    # Create the simulation
+    termination_condition = AllParticlesOffGridTerminationCondition()
+    sim = ParticleTracker(grid, termination_condition=termination_condition, seed=42)
+
+    # Load particles as a beam source collinear with the y-axis
+    nparticles = beam_energies.shape[0] * PARTICLES_PER_ENERGY
+
+    x0 = np.zeros(shape=(nparticles, 3))
+    v0 = np.zeros(shape=(beam_energies.shape[0], PARTICLES_PER_ENERGY, 3))
+
+    initial_speed = RelativisticBody(
+        particle=beam_particle, kinetic_energy=beam_energies
+    ).velocity
+
+    # The speed value should be copied into the y-component along the second
+    # axis
+    # TODO: figure out a cleaner way of doing this
+    for i, speed in enumerate(initial_speed):
+        v0[i, :, 1] = speed.si.value
+
+    v0 = np.reshape(v0, newshape=(nparticles, 3))
+    sim.load_particles(x0 * u.m, v0 * u.m / u.s, particle=beam_particle)
+
+    # Add stopping and scattering to the simulation
+    sim.add_stopping("NIST", target_material, target_density, NIST_material_string)
+    sim.add_scattering(
+        target_material,
+        target_density,
+        scatter_routine=Mott_Bethe_mean_squared_scattering,
```

Comment:
This only tests the MSR option - what about the differential cross section option?